### PR TITLE
Typescriptify charts

### PR DESCRIPTION
**Pull request by hwbllmnn**
@terrestris/devs Please review.

### CODE REVIEW

File: src/AxesUtil/AxesUtil.ts [hunk 1]
```diff
@@ -10,13 +9,46 @@ import {
...
+  max?: any;
+  harmonize?: boolean;
+  factor?: number;
+  orientation: 'x' | 'y';
+  gridWidth?: string;
+  gridColor?: string;
+  showGrid?: boolean;
+  gridOpacity?: number;
+  sanitizeLabels?: boolean;
+}
```
Comment (KaiVolland) [inline src/AxesUtil/AxesUtil.ts]: Are there any required Options?

File: src/BaseUtil/BaseUtil.ts [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+}
+
+export interface TitleConfiguration {
+  title?: string;
+  size?: [number, number];
+  position?: [number, number];
+  titlePadding?: string|number;
+  titleSize?: number;
+  titleColor?: string;
+}
```
Comment (KaiVolland) [inline src/BaseUtil/BaseUtil.ts]: Are there any required Params?

File: src/LegendComponent/LegendComponent.ts [hunk 3]
```diff
@@ -8,16 +10,37 @@ const SVG_DEFS = {
...
 
+interface LegendItemConfiguration {
+  type?: string;
+  style?: object;
+  title?: string;
+  tooltip?: string;
+  contextmenuHandler?: (event: any) => any;
+  onClick?: (event: any) => any;
+  customRenderer?: (legend: NodeSelection) => any;
+}
```
Comment (KaiVolland) [inline src/LegendComponent/LegendComponent.ts]: Are there any required Params?

File: package.json [hunk 4]
```diff
@@ -2,17 +2,17 @@
   "name": "@terrestris/d3-util",
   "version": "0.5.0",
   "description": "A set of helper classes for working with d3 charts.",
-  "main": "src/index.js",
+  "main": "src/index.ts",
```
Comment (KaiVolland) [inline package.json]: should point to `"dist/index.js"`

File: src/AxesUtil/AxesUtil.ts (hunk 1 above)
Comment (hwbllmnn) [reply src/AxesUtil/AxesUtil.ts]: Not really. The code was written to cope even with empty configuration objects in mind, that's why they're all optional.

File: src/LegendComponent/LegendComponent.ts (hunk 3 above)
Comment (hwbllmnn) [reply src/LegendComponent/LegendComponent.ts]: See above.

File: package.json (hunk 4 above)
Comment (hwbllmnn) [reply package.json]: Thanks, done in 1548ddb

File: src/BaseUtil/BaseUtil.ts (hunk 2 above)
Comment (hwbllmnn) [reply src/BaseUtil/BaseUtil.ts]: See above.

**Pull request merged**